### PR TITLE
PUPIL-586 feat(pupil header): create pupil journey header and story

### PR DESCRIPTION
**Pull request by benprotheroe**
# How to review this PR

_Leave this text block for the reviewer_

- Check [component hierarchy](https://miro.com/app/board/uXjVNnKBgyk=/?share_link_id=59445593794) is followed correctly
- Check the design [Heuristics](https://components.thenational.academy/?path=/docs/docs-howtodesigncomponents--docs#heuristics-for-component-design) have been followed
- Check [naming conventions](https://components.thenational.academy/?path=/docs/docs-namingconventions--docs) have been applied
- Check for these gotchyas:
  - Missing exports for Oak components
  - Accidental export of Internal components
  - Snapshots of unexpected components have been modified
  - Circular dependencies
  - Code duplication (via not using base components)
  - Non-functional storybook for this or related components
  - Sensitve files changed eg. atoms, or style tokens
  - Relative imports
  - Default exports

# Add your PR description below

Ia the header component for the pupil journey; which contains the icon, title and breadcrumb

it can be used for unit, tier and lesson listing for both primary and secondary

## Link to the design doc

https://www.figma.com/file/zqpeecgzMGKRli2gUdJOLp/Pupil-Browse?type=design&node-id=3993%3A50489&mode=dev

## A link to the component in the deployment preview

https://deploy-preview-143--lively-meringue-8ebd43.netlify.app/?path=/docs/components-organisms-pupil-oakpupiljourneyheader--docs

## Testing instructions

Go to deployment review and check that the component has the correct styles and can meet the needs for all its use cases

### CODE REVIEW

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ *
+ */
+export const OakPupilJourneyHeader = ({
+  iconBackground = "primary",
+  ...props
+}: OakPupilJourneyHeaderProps) => {
+  return (
+    <OakFlex $flexDirection={"row"} $gap={"space-between-m"} {...props}>
+      <OakHandDrawnCardWithIcon
```
Comment (benprotheroe) [inline src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx:30]: i didnt use the OakSubjectIcon component as the dimentions were different; and the breakpoints set didnt match the use in this title component

**Review by rightsaidjames — APPROVED** [review]

**Comment by messmike** [comment]
@benprotheroe I think this is looking good. I've mapped this out across the different breakpoints and also added an optional title that would appear for units with options: [](https://www.figma.com/file/YcWQMMhHPVVmc47cHHEEAl/Oak-Design-Kit?type=design&node-id=8311%3A11608&mode=dev)

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import React from "react";
+
+import { OakFlex, OakHeading, OakIconProps } from "../../../atoms";
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx:3]: can you clean up your relative imports :)

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+ * the icon, title and list of items are passed as props and change change depending on which page it is called
+ *
+ *
+ */
+export const OakPupilJourneyHeader = ({
+  iconBackground = "primary",
+  ...props
+}: OakPupilJourneyHeaderProps) => {
+  return (
+    <OakFlex $flexDirection={"row"} $gap={"space-between-m"} {...props}>
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx:29]: I'm not sure spreading props is wise here. There's a lot of unused values here which could end up causing unexpected behaviours.

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+  OakBulletListProps,
+  OakHandDrawnCardWithIcon,
+} from "../../../molecules";
+
+export type OakPupilJourneyHeaderProps = {
+  title: string;
+  iconName: OakIconProps["iconName"];
+  alt?: OakIconProps["alt"];
+  iconBackground?: "primary" | "secondary";
+} & OakBulletListProps;
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx:15]: given this is a specialised component , I think we could rename listItems to something more specific. Eg. breadcrumbs ?

File: src/components/organisms/pupil/OakPupilJourneyHeader/index.ts [hunk 5]
```diff
@@ -0,0 +1 @@
+export * from "./OakPupilJourneyHeader.js";
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyHeader/index.ts:1]: .js ???

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx (hunk 1 above)
Comment (kimon-satan) [reply src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx:30]: Fair enough, but woulkd it work now that Mike has added breakpoints. No worries if not but I guess you need to implement his changes in anycase

**Review by kimon-satan — CHANGES_REQUESTED** [review]
A few minor changes from me, but I think you need to implement Mike's responsive additions in anycase.

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx (hunk 3 above)
Comment (benprotheroe) [reply src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx:29]: ive changed it to flexProps; i thin its useful to be able to pass props down to the parent flex container at the page level

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx [hunk 6]
```diff
@@ -0,0 +1,61 @@
+import React from "react";
+
+import {
+  OakBulletList,
+  OakBulletListProps,
+  OakHandDrawnCardWithIcon,
+} from "../../../molecules";
```
Comment (kimon-satan) [inline src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx:7]: still relative

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx (hunk 3 above)
Comment (kimon-satan) [reply src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx]: It's improved but I still don't like it as an interface design. 2 props are exposed but are actually hardcoded (violates only expose props which are variable) and it's ambiguous as to what all of those generic flex props control. Instead pick the one or two props that you think the user actually needs to adjust at page level and keep the rest internal.

**Review by kimon-satan — CHANGES_REQUESTED** [review]
I still have a problem with flexProps. By exposing gap and flexdirection it violates 'only expose props which are variable' . But generally it goes against our principle of only exposing those props which are needed. Personally I wouldn't expose any flex props but if you have to pick the one or two that you think are needed and rename them (eg. containerFlexPa to make them less ambigous.)

**Comment by messmike** [comment]
Hey @benprotheroe give me a nudge when you want a design review for this component.

File: src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx (hunk 3 above)
Comment (benprotheroe) [reply src/components/organisms/pupil/OakPupilJourneyHeader/OakPupilJourneyHeader.tsx]: ok removed

**Comment by messmike** [comment]
Thanks @benprotheroe, feedback below:
- It looks like there may be a missing breakpoint for mobile (the OakSubjectIcon changes size for mobile from the tablet breakpoint).
- Only other thing was the option to have a unit title as a boolean that can be turned on to appear above the lesson or unit title for when there is an optional unit (or is that part of another ticket?)
Screenshot 2024-04-10 at 12 00 33 pm

**Review by messmike — APPROVED** [review]
Optionality title will be added in another ticket.

**Review by kimon-satan — APPROVED** [review]

**Pull request merged**